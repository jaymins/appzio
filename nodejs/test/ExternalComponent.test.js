const ExternalComponent = require('../src/components/ExternalComponent')
const { parseObject } = require('../utils')

describe('ExternalComponent', () => {
  it('Should create an object with phpComponent type', () => {
    const component = ExternalComponent({ component: 'getComponentText' })
    expect(component.type).toEqual('phpComponent')
  })
  it('Should create an object depending on the passed object', () => {
    const params = {
      variable: 'var'
    }

    const component = ExternalComponent({
      component: 'getComponentText',
      content: 'This is some sample text',
      params: params
    })

    expect(component.content).toEqual('This is some sample text')
    expect(component.params).toEqual(params)
  })
  it('Should parse style object to be understandable by the device', () => {
    const style = {
      backgroundColor: '#ffffff',
      color: '#000000',
      fontSize: 14
    }

    const component = ExternalComponent({
      component: 'getComponentText',
      content: 'This is some sample text',
      style
    })

    expect(component.style).toEqual(parseObject(style))
  })
  it('Should match a working snapshot', () => {
    const component = ExternalComponent({
      component: 'getComponentFormFieldText',
      content: '',
      params: {
        hint: 'First Name'
      },
      style: {
        backgroundColor: '#ffffff'
      }
    })

    expect(component).toMatchSnapshot()
  })
})
