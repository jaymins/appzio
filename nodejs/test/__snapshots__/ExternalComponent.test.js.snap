// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalComponent Should match a working snapshot 1`] = `
Object {
  "component": "getComponentFormFieldText",
  "content": "",
  "params": Object {
    "hint": "First Name",
  },
  "style": Object {
    "background-color": "#ffffff",
  },
  "type": "phpComponent",
}
`;
